subliminal/providers/legendastv.py

```diff
@@ -0,0 +1,466 @@
+
+        return subtitles
+
+    def list_subtitles(self, video, languages):
+        """
+        Returns a list of subtitles for the defined video and requested languages
+
+        :param video:
+        :param languages: the requested languages
+        :return: a list of subtitles for the requested video and languages
+        :rtype : ``list`` of ``LegendasTvSubtitles``
+        """
+        if isinstance(video, Episode):
+            params = {'type': 'episode', 'series': video.series, 'season': video.season,
+                      'episodeNumber': video.episode, 'year': video.year}
+        elif isinstance(video, Movie):
+            params = {'type': 'movie', 'title': video.title, 'year': video.year}
+
+        return [s for l in languages for s in self.query(l, video, params) if params]
+
+    def get_subtitle_names(self, subtitle_id, timestamp):
+        """
+        Returns all subtitle names for a specific subtitle_id. Only subtitle names are returned.
+
+        :param subtitle_id: the id used to download the compressed file
+        :param timestamp: represents the last update timestamp of the file
+        :return: list of subtitle names
+        :rtype: ``list`` of ``string``
+        """
+        return self._uncompress(
+            subtitle_id, timestamp,
+            lambda cf: [f for f in cf.namelist()
+                        if 'legendas.tv' not in f.lower() and f.lower().endswith(SUBTITLE_EXTENSIONS)])
+
+    def extract_subtitle(self, subtitle_id, subtitle_name, timestamp):
+        """
+        Extract the subtitle content from the compressed file. The file is downloaded, the subtitle_name is uncompressed
+        and its contents is returned.
+
+        :param subtitle_id: the id used to download the compressed file
+        :param subtitle_name: the filename to be extracted
+        :param timestamp: represents the last update timestamp of the file
+        :return: the subtitle content
+        :rtype : ``string``
+        """
+        return self._uncompress(subtitle_id, timestamp,
+                                lambda cf, name: fix_line_ending(cf.read(name)), subtitle_name)
+
+    def _uncompress(self, subtitle_id, timestamp, function, *args, **kwargs):
+        content = self.download_content(subtitle_id, timestamp)
+
+        # Download content might be a rar file (most common) or a zip.
+        # Unfortunately, rarfile module only works with files (no in-memory streams)
+        tmp = NamedTemporaryFile()
+        try:
+            tmp.write(content)
+            tmp.flush()
+
+            cf = RarFile(tmp.name) if is_rarfile(tmp.name) else (ZipFile(tmp.name) if is_zipfile(tmp.name) else None)
+
+            return function(cf, *args, **kwargs) if cf else None
+        finally:
+            tmp.close()
+
+    @region.cache_on_arguments(expiration_time=EPISODE_EXPIRATION_TIME)
+    def download_content(self, subtitle_id, timestamp):
```

Comment:
subliminal stores its cache into `~/.config`, which is a bad idea because it should be used for **permanent** user **configuration** files. Ever-growing, **safely deletable** cache directories should be stored in `~/.cache` instead.